File: stripe/src/main/java/com/stripe/android/SetupIntentResult.java
package com.stripe.android;

import androidx.annotation.NonNull;

import com.stripe.android.model.SetupIntent;

public class SetupIntentResult extends StripeIntentResult<SetupIntent> {

    private SetupIntentResult(@NonNull Builder builder) {
        super(builder.mSetupIntent, builder.mOutcome);
    }

    static final class Builder implements ObjectBuilder<SetupIntentResult> {
        private SetupIntent mSetupIntent;
        @Outcome private int mOutcome;

        @NonNull
        Builder setSetupIntent(@NonNull SetupIntent setupIntent) {
            mSetupIntent = setupIntent;
            return this;
        }

        @NonNull
        Builder setOutcome(@Outcome int outcome) {
            mOutcome = outcome;
            return this;
        }

        @NonNull
        public SetupIntentResult build() {
            return new SetupIntentResult(this);
        }
    }
}

File: stripe/src/main/java/com/stripe/android/SetupIntentResult.kt
package com.stripe.android

import com.stripe.android.model.SetupIntent
import kotlinx.android.parcel.Parcelize

/**
 * A model representing the result of a [SetupIntent] confirmation via [Stripe.confirmSetupIntent]
 * or handling of next actions via [Stripe.handleNextActionForSetupIntent].
 */
@Parcelize
data class SetupIntentResult internal constructor(
    override val intent: SetupIntent,
    @Outcome private val outcomeFromFlow: Int = 0,
    val failureMessage: String? = null
) : StripeIntentResult<SetupIntent>(outcomeFromFlow)
